contracts/gnosis/EIP4337Manager.sol

```diff
@@ -0,0 +1,179 @@
+//SPDX-License-Identifier: GPL
+pragma solidity ^0.8.7;
+
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@gnosis.pm/safe-contracts/contracts/GnosisSafe.sol";
+import "./EIP4337Fallback.sol";
+import "../EntryPoint.sol";
+    using ECDSA for bytes32;
+
+/**
+ * Main EIP4337 module.
+ * Called (through the fallback module) using "delegate" from the GnosisSafe as an "IWallet",
+ * so must implement validateUserOp
+ * holds an immutable reference to the EntryPoint
+ * Inherits GnosisSafeStorage so that it can reference the memory storage
+ */
+contract EIP4337Manager is GnosisSafe, IWallet {
+
+    address public immutable eip4337Fallback;
+    address public immutable entryPoint;
+
+    constructor(address anEntryPoint) {
+        entryPoint = anEntryPoint;
+        eip4337Fallback = address(new EIP4337Fallback(address(this)));
+    }
+
+    /**
+     * delegate-called (using execFromModule) through the fallback, so "real" msg.sender is attached as last 20 bytes
+     */
+    function validateUserOp(UserOperation calldata userOp, bytes32 requestId, uint256 missingWalletFunds) external override {
+        address _msgSender = address(bytes20(msg.data[msg.data.length - 20 :]));
```

Comment:
You are not checking the `msg.sender` here at all.
You must check that you have come here from the `EIP4337Fallback` instance. Otherwise the `_msgSender` may be faked.